src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/security-recommendations/SecurityRecommendationsLayout.tsx

```diff
@@ -58,12 +63,64 @@ export function SecurityRecommendationsLayout(
   // The non-null assertion here should be safe since we already did this check
   // in `./[type]/page.tsx`:
   const { title, illustration, content, exposedData } = pageData!;
+  const hasExposedData = exposedData.length > 0;
+
+  // TODO: Write unit tests MNTOR-2560
+  /* c8 ignore start */
+  const handlePrimaryButtonPress = async () => {
+    const securityRecommendatioBreachClasses: Record<
+      SecurityRecommendationTypes,
+      | (typeof SecurityRecommendationDataTypes)[keyof typeof SecurityRecommendationDataTypes]
+      | null
+    > = {
+      email: SecurityRecommendationDataTypes.Email,
+      phone: SecurityRecommendationDataTypes.Phone,
+      ip: SecurityRecommendationDataTypes.IP,
+      done: null,
+    };
+
+    const dataType = securityRecommendatioBreachClasses[props.type];
+    // Only attempt to resolve the breaches if the following conditions are true:
+    // - There is a matching data class type in this step
+    // - The current step has unresolved exposed data
+    // - There is no pending breach resolution request
+    if (!dataType || !hasExposedData || isResolving) {
+      return;
+    }
+
+    setIsResolving(true);
+    try {
+      const response = await fetch("/api/v1/user/breaches/bulk-resolve", {
```

Comment:
There is a good amount of logic that is specific to the bulk resolution of breaches: Setting component state, redirecting users depending on the next step, and guarding the resolution request against being repeatedly called. The only part that I’d be really moving over would be the actual `fetch` call or otherwise, we’d need to pass a lot of params, which I’m not sure would make things cleaner.